### DISCUSSION TITLE
can not use prebuilt artifacts

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#4661] [General]
https://github.com/xmake-mirror/build-artifacts/releases 计算的build hash和本地xmake的hash不一样了？本地使用add_requires引入的包都不会走预编译了

**Comment by waruqi** [comment]
可能是 runtimes 那个 patch 引入的问题，等下周我看下，你可以暂时先回退之前的版本

**Comment by xq114** [comment]
找到问题了，清空所有package文件夹下的hash文件夹成功了。似乎有原来的包hash就不会走新的hash计算路线

https://github.com/xmake-io/xmake/blob/445e43b40846b29b9abb1293b32b27b7104f54fa/xmake/core/package/package.lua#L1625-L1633

**Reply by xq114** [reply]
而且需要更新到dev的xmake

**Reply by waruqi** [reply]
> 似乎有原来的包hash就不会走新的hash计算路线

xmake/dev 兼容了老的 vs_runtime 的 hash ，如果本地安装了老的包，不会触发新的安装。。

- 如果是新构建的包，是 ci 上 build_artifacts 用了新的 runtimes hash ，导致云端是 新 hash，如果用的是 xmake/dev 应该能命中 
- 如果是老构建的包，云端的 hash 应该是老的带 vs_runtime 的，如果用老的 xmake 应该能命中，xmake/dev 无法命中云端的，但能兼容命中本地的

**Reply by waruqi** [reply]
另外有原来的包存在，不应该直接用现有的本地包么，为啥还要走云端拉取呢？
